scheduler/src/cook/scheduler/constraints.clj

```diff
@@ -89,13 +89,19 @@
     [this _ vm-attributes _]
     (job-constraint-evaluate this _ vm-attributes)))
 
+(defn job->previous-hosts-to-avoid
+  "Given a job, returns the set of hostnames the
+  job's instances have run on (except preemptions)."
+  [job]
+  (->> (:job/instance job)
+       (remove #(true? (:instance/preempted? %)))
+       (mapv :instance/hostname)))
```

Comment:
We need to make this a set before returning.

Reply:
Added call to `distinct`